docs/src/references/c8y-configuration-management.md

```diff
@@ -163,3 +279,232 @@ Note that:
+      or `$TEDGE_HTTP_ROOT/$CHILD_DEVICE_ID/config_snapshop/c8y-configuration-plugin`
+      (depending on the file has been pushed by the cloud or the child device).
+   1. The `c8y_configuration_plugin` puts this copy under
+      `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID/c8y-configuration-plugin.toml`,
+      and starts to used it for an updated list of configuration files
+      (to which this file `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID/c8y-configuration-plugin.toml`
+      is implicitly added with type `c8y-configuration-plugin`).
+   1. If the request were triggered by the cloud,
+      the `c8y_configuration_plugin` proceeds on processing the `successful` status message,
+      notifying the cloud and removing the file `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID/c8y-configuration-plugin.toml`.
+      Note that the request might have been triggered by the main device on start,
+      or by the child device itself (see below),
+      in that case the cloud most not be notified. 
+1. On start, the `c8y_configuration_plugin` check for each child device `$CHILD_DEVICE_ID`,
+   listed in the directory `$TEDGE_CONFIG_DIR/c8y/`,
+   if there is a known `$CHILD_DEVICE_ID/c8y-configuration-plugin.toml`.
+   1. If the configuration is missing for some `$CHILD_DEVICE_ID`
+      (i.e. if the directory `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID` exists
+      but not the file `$TEDGE_CONFIG_DIR/c8y/$CHILD_DEVICE_ID/c8y-configuration-plugin.toml`).
+   1. Then the `c8y_configuration_plugin` request this file from the child device
+      by sending an MQTT message on the topic `tedge/$CHILD_DEVICE_ID/commands/req/config_snapshot`.
+   1. The payload of this message is a JSON record with 2 required fields:
+       * `"url": "http://$TEDGE_HTTP/tedge/$CHILD_DEVICE_ID/config_snapshot/c8y-configuration-plugin"`
+       * `"type": "c8y-configuration-plugin"`    
+1. The child-device configuration software has nothing specific to do on requests
+   that are related to the `"type": "c8y-configuration-plugin"`.
+   However, on start and on local changes of the list of configuration files,
+   The child-device configuration software should push to the thin-edge device
+   this new or updated content. This is done as a regular snapshot request.
+      1. `PUT`s the content of updated `$CHILD_DEVICE_ID/c8y-configuration-plugin.toml`
+         to the `http://$TEDGE_HTTP/tedge/$CHILD_DEVICE_ID/config_snapshot/c8y-configuration-plugin`.
+      1. On success of the upload, the child-device configuration software
+         notifies the `c8y_configuration_plugin` with an MQTT message published on the topic
+         `tedge/$CHILD_DEVICE_ID/commands/res/config_snapshot`
+      1. The payload is a JSON record with 2 required fields:
+        * `"status": "successful"`
+        * `"type": "c8y-configuration-plugin"`
+      1. Note that there is no need to send error or executing notification messages here,
+         since this is a spontaneous operation.
```

Comment:
This section talks about the handling of "config list" for a child device during the startup of the plugin. How the config list of a new child device added dynamically, while the plugin is already running is not explicitly specified, but I'm assuming this is how it works.

1. When the new child device is added, the user run `c8y-config-plugin --init <CHILD_DEVICE_ID>` command which creates the directory `/etc/tedge/c8y/$CHILD_DEVICE_ID` for that child.
2. The inotify mechanism of the plugin detects the creation of this new directory for the new child device and initiates the config_snapshot request for `"type": "c8y-configuration-plugin"` as described above.

@didier-wenzek Could you please confirm if that's what you had in mind to handle dynamic addition of child devices?

Reply:
I was not mindfully thinking to that scenario but these are indeed a correct consequence of what has been written.

That said, I'm not happy with this complex dance with a mix of manual operation, inotify and MQTT.

@albinsuresh, @rina23q and @initard : I think we need to consider to simplify these initialization steps with the child device sending its `c8y-configuration-plugin.toml` file on start.

Reply:
The initialization steps have been simplified with the child device sending its c8y-configuration-plugin.toml file on start.